tools/functional-tester/etcd-tester/lease_stresser.go

```diff
@@ -258,28 +258,35 @@ func (ls *leaseStresser) getLeaseByID(ctx context.Context, leaseID int64) (*pb.L
 }
 
 func (ls *leaseStresser) hasLeaseExpired(ctx context.Context, leaseID int64) (bool, error) {
-	resp, err := ls.getLeaseByID(ctx, leaseID)
-	plog.Debugf("hasLeaseExpired %v resp %v error (%v)", leaseID, resp, err)
-	if rpctypes.Error(err) == rpctypes.ErrLeaseNotFound {
-		return true, nil
+	for ctx.Err() == nil {
```

Comment:
this for condition seems strange. how about checking the ctx.Err at the end of the loop instead?